src/lib/CurrencyInput.svelte

```diff
@@ -147,13 +155,16 @@
 		}
 	};
 
-	const setFormattedValue = (hasMinFractionDigits?: boolean) => {
+	const setFormattedValue = () => {
+		// Do nothing because the page hasn't mounted yet
+		if (!dom) return;
+
 		// Previous caret position
-		const startCaretPosition = inputTarget?.selectionStart || 0;
+		const startCaretPosition = inputElement?.selectionStart || 0;
 		const previousFormattedValueLength = formattedValue.length;
 
 		// Apply formatting to input
-		formattedValue = isZero && !isZeroNullish ? '' : formatCurrency(value, fractionDigits, hasMinFractionDigits ? fractionDigits : 0);
+		formattedValue = isZero && !isZeroNullish ? '' : formatCurrency(value, fractionDigits, dom.activeElement == inputElement ? 0 : fractionDigits);
```

Comment:
If `dom.activeElement == inputElement` is `true` it means the user is typing on it so we don't want to automatically format with fraction digits.

Otherwise, we format with digits because the user clicked out of the input or the value has changed from somewhere else (which fixes the bug described in #74).